dashboard/app/models/experiment.rb

```diff
@@ -35,46 +35,30 @@ class Experiment < ApplicationRecord
 
   @@experiments = nil
 
-  def self.should_cache?
-    Script.should_cache?
-  end
-
-  def self.get_all_enabled(user: nil, section: nil, script: nil)
-    if Experiment.should_cache?
-      Experiment.get_all_enabled_cached(user: user, section: section, script: script)
-    else
-      Experiment.get_all_enabled_uncached(user: user, section: section, script: script)
-    end
-  end
-
-  def self.get_all_enabled_uncached(user: nil, section: nil, script: nil)
-    experiments = Experiment.descendants.map do |experiment_type|
-      experiment_type.get_enabled(user: user, section: section, script: script)
-    end
-    experiments = experiments.reduce(Experiment.none) do |e1, e2|
-      e1.or(e2)
-    end
-    now = DateTime.now
-    experiments.where('start_at IS NULL OR start_at < ?', now).
-      where('end_at IS NULL OR end_at > ?', now).
-      where('script_id IS NULL OR script_id = ?', script.try(:id))
-  end
-
-  def self.get_all_enabled_cached(user: nil, section: nil, script: nil)
+  def self.get_all_enabled(user: nil, section: nil, script: nil, experiment_name: nil)
     if @@experiments.nil? || @@experiments_loaded < DateTime.now - MAX_CACHE_AGE
-      Experiment.update_cache
+      update_cache
```

Comment:
This means we will repeatedly update the cache if there are no experiments (regardless of cache age)? Though perhaps an edge case that shouldn't happen on `production`, this seems inefficient for other environments (e.g., `levelbuilder`, which currently has no experiments in the DB).

Why do we care whether `@@experiments.nil?`?